src/backend/api/Fusion.Resources.Api/Controllers/Validation/CustomValidatorExtensions.cs

```diff
@@ -178,6 +178,54 @@ public static string ToLowerFirstChar(this string input)
 
         public static string JsPropertyName(this CustomContext context) => context.PropertyName.ToLowerFirstChar();
 
+
+        /// <summary>
+        /// The preferred contact mail should try to weed out main private mail domains.
+        /// </summary>
+        public static IRuleBuilderOptions<T, string?> NotHaveInvalidMailDomain<T>(this IRuleBuilder<T, string?> ruleBuilder)
+        {
+            
+            return ruleBuilder.SetValidator(new EmailDomainValidator())
+                .WithMessage("Mail domain should not be private. Most major private mail domains are rejected (gmail, hotmail, icloud etc.)");
+        }
     }
 
+    public class EmailDomainValidator : PropertyValidator, IPropertyValidator
```

Comment:
Just asking. Do we have a "global" email validator somewhere that you are aware of ?

Reply:
We have a validator that checks that the email is a valid email. Think thats in the Web library. However this is validating for illegal domains in the email...